### PR TITLE
Manufacturing skills utility

### PR DESCRIPTION
**Pull request by iaincrawf**

### CODE REVIEW

File: src/Facade/ResourceBuilders/ManufacturingSkillResourceBuilder.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
+﻿using Linn.Production.Domain.LinnApps;
+using Linn.Production.Resources;
```
Comment (richardiphillips) [inline src/Facade/ResourceBuilders/ManufacturingSkillResourceBuilder.cs]: usings should be inside namespace. Is your style cop plugin still working. I had to reinstall mine after one on the Visual Studio / resharper updates

File: src/Facade/ResourceBuilders/ManufacturingSkillsResourceBuilder.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
+﻿using Linn.Production.Domain.LinnApps;
```
Comment (richardiphillips) [inline src/Facade/ResourceBuilders/ManufacturingSkillsResourceBuilder.cs]: again there are a few of these that should be inside namespace.

File: src/Facade/Services/ManufacturingSkillService.cs [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+using Linn.Common.Facade;
+using Linn.Common.Persistence;
+using Linn.Production.Domain.LinnApps;
+using Linn.Production.Resources;
+
+namespace Linn.Production.Facade.Services
+{
+    public class ManufacturingSkillService : FacadeService<ManufacturingSkill, string, ManufacturingSkillResource, ManufacturingSkillResource>
+    {
+       // private readonly IRepository<ManufacturingSkill, int> manufacturingSkillRepository;
```
Comment (richardiphillips) [inline src/Facade/Services/ManufacturingSkillService.cs]: remove the comments?

File: src/Facade/Services/ManufacturingSkillService.cs [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+        public ManufacturingSkillService(IRepository<ManufacturingSkill, string> repository, ITransactionManager transactionManager) : base(repository, transactionManager)
+           // , IRepository<ManufacturingSkill, int> manufacturingSkillRepository) 
+        {
+           // this.manufacturingSkillRepository = manufacturingSkillRepository;
+        }
+
+        public IResult<ManufacturingSkill> CreateManufacturingSkill(ManufacturingSkillResource resource)
```
Comment (richardiphillips) [inline src/Facade/Services/ManufacturingSkillService.cs]: why do you need these extra methods? I think most of this should be taken care of by the library code unless you've got a special case (which you might). Have a look at the ate fault code implementation

File: src/Facade/Services/ManufacturingSkillService.cs [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        }
+
+        protected override ManufacturingSkill CreateFromResource(ManufacturingSkillResource resource)
+        {
+           return new ManufacturingSkill(resource.SkillCode, resource.Description, resource.HourlyRate);
+        }
+
+        protected override void UpdateFromResource(ManufacturingSkill entity, ManufacturingSkillResource updateResource)
+        {
+            entity.SkillCode = updateResource.SkillCode;
```
Comment (richardiphillips) [inline src/Facade/Services/ManufacturingSkillService.cs]: could do with some tests for the facade service too.

File: src/Domain.LinnApps/ManufacturingSkill.cs [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+    public class ManufacturingSkill
+    {
+        public string SkillCode { get; set; }
+
+        public string Description { get; set; }
+
+        public int? HourlyRate { get; set; }
+
+        public ManufacturingSkill(string skillCode, string description, int? hourlyRate)
+        {
```
Comment (richardiphillips) [inline src/Domain.LinnApps/ManufacturingSkill.cs]: I think the constructor should be at the top. You might need a private parameterless one for entity framework too

File: src/Persistence.LinnApps/ServiceDbContext.cs [hunk 7]
```diff
@@ -27,6 +27,9 @@ public class ServiceDbContext : DbContext
 
         public DbQuery<WhoBuiltWhat> WhoBuiltWhat { get; set; }
 
+        public DbSet<ManufacturingSkill> ManufacturingSkills { get; set; }
+
+
```
Comment (richardiphillips) [inline src/Persistence.LinnApps/ServiceDbContext.cs]: double blank line

File: src/Service.Host/client/src/containers/manufacturingSkills/ManufacturingSkills.js [hunk 8]
```diff
@@ -0,0 +1,25 @@
+import { connect } from 'react-redux';
+import { fetchErrorSelectors } from '@linn-it/linn-form-components-library';
+import initialiseOnMount from '../initialiseOnMount';
```
Comment (richardiphillips) [inline src/Service.Host/client/src/containers/manufacturingSkills/ManufacturingSkills.js]: should this not use the initialiseOnMount from shared library?

File: src/Service/Modules/ManufacturingSkillsModule.cs [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+    public sealed class ManufacturingSkillsModule : NancyModule
+    {
+        private readonly IFacadeService<ManufacturingSkill, string, ManufacturingSkillResource, ManufacturingSkillResource> manufacturingSkillService;
+
+        public ManufacturingSkillsModule(IFacadeService<ManufacturingSkill, string, ManufacturingSkillResource, ManufacturingSkillResource> manufacturingSkillService)
+        {
+            this.manufacturingSkillService = manufacturingSkillService;
+            this.Get("/production/manufacturing-skills", _ => this.GetAll());
+            this.Get("production/manufacturing-skills/{skillCode*}", parameters => this.GetById(parameters.skillCode));
+            this.Put("production/manufacturing-skills/{skillCode*}", parameters => this.UpdateManufacturingSkill(parameters.skillCode));
```
Comment (richardiphillips) [inline src/Service/Modules/ManufacturingSkillsModule.cs]: this uri won't work unfortunately.  Remember we're limited to 
- /production/reports*
           - /production/works-orders*
           - /production/quality*
           - /production/resources*
           - /production/maintenance* 
at the moment.  Resources could be the one to go for

File: src/Facade/Services/ManufacturingSkillService.cs (hunk 4 above)
Comment (iaincrawf) [reply src/Facade/Services/ManufacturingSkillService.cs]: Ah you're right, I forgot to undo adding these methods and remove the other related code I had commented out

File: src/Service/Modules/ManufacturingSkillsModule.cs (hunk 9 above)
Comment (iaincrawf) [reply src/Service/Modules/ManufacturingSkillsModule.cs]: completely forgot about this, will put it in resources!

**Pull request merged**